Assignment 3.Rmd

```diff
@@ -127,29 +129,119 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
 
+```{r}
+load("p.c4.Rdata")
+# load cleaned data saved from assignment2
+t1 <- scale(p.c4)
+fit2 <- kmeans(t1,4)
+t2 <- data.frame(t1,fit2$cluster)
+# find cluster and label
+p.c5 <- as.matrix(p.c4)   
+# person-class matrix done
+
+p.p <- p.c5 %*% t(p.c5) 
+# matrix * matrix(T) to get person-person
+diag(p.p)<-NA   
+# make diagnosis as NA because that's pairing themselves
+# person-person matrix done
+
+library(igraph)
+g3<-graph.adjacency(p.p,mode="undirected")
+plot(g3, layout=layout.fruchterman.reingold, vertex.color=t2$fit2.cluster)  
+# graph the network with color degined by cluster
+
+
+```
 ##Part III
 
 In class activity 6 you clustered students in the class by the answers to a questionaire. Create a visualization that shows the overlap between these clusters and the clusters generated in part II.
 
+```{r}
+d1 <- read.csv("/Users/xiyang/Desktop/fall 2019/HUDK 4050/r-project/assignment/class-activity-6/HUDK405019-clustering.csv")
+# load original data from class activity 6
+d2 <- data.frame(lapply(d1, gsub, pattern="[[:alpha:]]", replacement=""))
+d2 <- data.frame(lapply(d2, gsub, pattern="°", replacement=""))
+d2 <- data.frame(lapply(d2, gsub, pattern=",", replacement=""))
+d2 <- data.frame(lapply(d2, gsub, pattern=" ", replacement=""))
+# replacing characters and symbols
+names <- matrix(paste(d1$First.Name,d1$Last.Name,sep = " "),50:1)
+row.names(d2) <- names
+# change the row names to students' names
+d3 <- d2[-c(1,10,14,20,22,42,50),]
+# delete invail data
+d4 <- as.data.frame(d3[,c(15,16)])
+colnames(d4) <- c("latitude", "longitude")
+d5 <- d4 %>% mutate_all(as.character) %>% mutate_all(as.numeric)
+# extract only latitude and longitude, data cleaning done
+
+d6 <- scale(d5)
+fit3 <- kmeans(d6,4)
+# cluster students in this data
+
+p6 <- as.data.frame(cbind(toupper(rownames(p.c5)),fit2$cluster))
+p6 <- p6 %>% mutate_all(as.character)
+d7 <- as.data.frame(cbind(toupper(rownames(d4)),fit3$cluster))
+colnames(d7) <- c("V1","V3")
+d7 <- d7 %>% mutate_all(as.character)
+t_final <- inner_join(p6,d7,by = "V1")
+# merge two table with cluster by the same names
+
+t_final <- cbind(1:28,t_final)
+colnames(t_final) <- c("No.","names","cluster1","cluster2")
+plot(x = t_final$No.,y = t_final$cluster1, xlab = "name", ylab = "cluster", main = "overlapping cluster", type = "n",xaxt = "n")
+points(x = t_final$No., y = t_final$cluster1, col = rgb(0.49,0.73,0.91), pch = 3)
+points(x = t_final$No., y = t_final$cluster2, col = rgb(0.95,0.61,0.73), pch = 4)
+text(seq(1.5,30), par("usr")[3]-0.35, labels = t_final$names, srt = -60, pos = 1, xpd = TRUE,cex = 0.6)
+legend("center", legend = c("cluster1", "cluster2"),
+           fill = c(rgb(0.49,0.73,0.91),rgb(0.95,0.61,0.73)))
+
+# cluster from part2 is labeled as "+" and cluster from part3 is labeled as "×"，so the overlap would be "*"
+# from the plot we can see there are some "*"
+
+```
```

Comment:
Overall, great job on the coding part of this assignment but you may want to work on the analysis aspect of it. Nonetheless, keep up the good work!